internal/daemon/worker/status.go

```diff
@@ -144,12 +144,15 @@ func (w *Worker) sendWorkerStatus(cancelCtx context.Context) {
 	statusCtx, statusCancel := context.WithTimeout(cancelCtx, common.StatusTimeout)
 	defer statusCancel()
 
+	keyId := w.WorkerAuthCurrentKeyId.Load()
```

Comment:
should we check if keyId is empty?

Reply:
It could be empty if it's a KMS worker and doesn't have a keyId, so I think this is fine- @jefferai ?

Reply:
This is correct in that it's empty if it's a KMS worker, but there's actually a bug -- not your bug, not my bug, just from two things happening in parallel -- it's also empty if we haven't (a) created a fetch request due to already having credentials and (b) haven't rotated. You probably need to update the logic in worker.go (starting around line 197)  so that if `!createFetchRequest` we derive and populate the key ID anyways.

I also think startListeners should go below that block instead of above -- it is totally fine where it is but it's sort of of a piece with startControllerConnections so I think a bit cleaner for them to be together.

Reply:
ty for updaing worker.go -> I'm curious if we should still detect if it's empty, since it's def an error condition now I believe.